### PR TITLE
Payouts

### PR DESCRIPTION
**Pull request by Ibrahim-Jarada**
Relates #63

### CODE REVIEW

File: src/controllers/users/getUserPayouts.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Request, Response, NextFunction } from 'express';
+import { getPayouts } from '../../services';
+import { constants, dto, errorMessages } from '../../helpers';
+
+export default async (request: Request, response: Response, next: NextFunction)
+: Promise<void> => {
+  const { httpStatus, messages } = constants;
+  const { user } = dto.generalDTO.userDTO(request);
```
Comment (RandSohail) [inline src/controllers/users/getUserPayouts.ts:8]: The same thing create new DTO and make whatever you want there, DON'T USE TOW

File: src/controllers/users/askPayoutRequest.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Request, Response, NextFunction } from 'express';
+import { askForPayout } from '../../services';
+import { constants, dto, errorMessages } from '../../helpers';
+
+export default async (request: Request, response: Response, next: NextFunction)
```
Comment (RandSohail) [inline src/controllers/users/askPayoutRequest.ts:5]: The request type you use in [this](https://github.com/NextUp-comedy/api-nc/blob/abf24f9870a2ef7b5b4a60e8495e78e6a937e320/src/helpers/dto/users.ts#L31) is an `IUserRequest`, I believe you should use  the same type in the controller

File: src/controllers/users/askPayoutRequest.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+: Promise<void> => {
+  const { httpStatus, messages } = constants;
+  const {
+    amount, createdBy, payoutStatusId, updatedBy, userId,
+  } = dto.usersDTO.askForPayoutDTO(request);
+  try {
+    if (+amount <= 0) throw errorMessages.NO_BALANCE;
```
Comment (RandSohail) [inline src/controllers/users/askPayoutRequest.ts:12]: Why is the amount in the [DTO ](https://github.com/NextUp-comedy/api-nc/blob/abf24f9870a2ef7b5b4a60e8495e78e6a937e320/src/helpers/dto/users.ts#L33) is a string then when using it you transfer it as a number?
why not transfer it as a number in the DTO?

**Review by RandSohail — CHANGES_REQUESTED** [review]

**Review by AymanSami98 — APPROVED** [review]

**Review by zaher-aa — APPROVED** [review]
الله أكبر عليك, ايه الحلاوة دي!

**Pull request merged**